pkg/controller/chaosengine/chaosengine_controller.go

```diff
@@ -572,3 +598,51 @@ func getChaosType() error {
 	}
 	return nil
 }
+
+// reconcileForDelete
+func (r *ReconcileChaosEngine) reconcileForDelete(request reconcile.Request) (reconcile.Result, error) {
+
+	r.recorder.Eventf(engine.Instance, corev1.EventTypeNormal, "Stopping reconcile for chaosEngine", "Will remove all Chaos resources for chaosEngineUUID: %v", string(engine.Instance.UID))
+	optsDelete := []client.DeleteAllOfOption{
+		client.InNamespace(request.NamespacedName.Namespace),
+		client.MatchingLabels{"engineUUID": string(engine.Instance.UID)},
+	}
+	reqLogger := chaosTypes.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	err := r.client.DeleteAllOf(context.TODO(), &corev1.Pod{}, optsDelete...)
+	if err != nil {
+		reqLogger.Error(err, "Unable to delete chaosEngine allocated Chaos Resources")
+		return reconcile.Result{}, err
+	}
+	opts := client.UpdateOptions{}
+	engine.Instance.Spec.EngineStatus = "stopped"
+	engine.Instance.ObjectMeta.Finalizers = utils.RemoveString(engine.Instance.ObjectMeta.Finalizers, "chaosengine.litmuschaos.io/finalizer")
+	err = r.client.Update(context.TODO(), engine.Instance, &opts)
+	if err != nil {
+		reqLogger.Error(err, "Unable to remove Finalizer from chaosEngine Resource")
```

Comment:
return the error

Reply:
Returning just below.

Reply:
Just returning from this function, might not be a good way because that function is performing some more function, and then handling the error without appropriate string attached would be a little vague.
What is the reason for not logging here?

Reply:
It's a kind of re-printing the error two times the reconcile function will also print the error as well as reconcileForDelete function